src/artm/core/master_component.cc

```diff
@@ -412,10 +412,74 @@ void MasterComponent::ImportModel(const ImportModelArgs& args) {
     BOOST_THROW_EXCEPTION(CorruptedMessageException("Unable to read from " + args.file_name()));
 
   instance_->SetPhiMatrix(args.model_name(), target);
-  LOG(INFO) << "Import completed, token_size = " << target->token_size()
+  LOG(INFO) << "Import of model completed, token_size = " << target->token_size()
     << ", topic_size = " << target->topic_size();
 }
 
+void MasterComponent::ExportScoreTracker(const ExportScoreTrackerArgs& args) {
+  if (boost::filesystem::exists(args.file_name()))
+    BOOST_THROW_EXCEPTION(DiskWriteException("File already exists: " + args.file_name()));
+
+  std::ofstream fout(args.file_name(), std::ofstream::binary);
+  if (!fout.is_open())
+    BOOST_THROW_EXCEPTION(DiskWriteException("Unable to create file " + args.file_name()));
+
+  LOG(INFO) << "Exporting score tracker to " << args.file_name();
+
+  const char version = 0;
+  fout << version;
+
+  // We expect here that each ScoreData object has suitable size (< 2GB)
+  for (auto& item : instance_->score_tracker()->GetDataUnsafe()) {
+    auto str = item->SerializeAsString();
+    fout << str.size();
```

Comment:
Ok, then it'll be better to somehow track these pieces of ... self-written code.